.buildkite/pipeline.yml

```diff
@@ -91,7 +91,7 @@ steps:
   - label: ":docker: Publish docker image"
     key: "publish"
     command: ".buildkite/scripts/build_push_docker_image.sh"
-    if: build.env("BUILDKITE_PULL_REQUEST") == null
+    if: build.env("BUILDKITE_PULL_REQUEST") == ''
```

Comment:
Here, it would be better to just run this step if it is branch main. Could that be used BUILDKITE_BRANCH for that ?